app/controllers/posts_controller.rb

```diff
@@ -2,23 +2,41 @@
 
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    top_post_id = Post.joins(:votes).group(:post_id).count.sort.to_h
+    @top_post = if top_post_id.nil?
+                  Post.find(top_post_id.first[0])
+                else
+                  Post.last
+                end
+    @posts = Post.where.not(id: top_post_id)
+
   end
 
   before_action :authenticate_author!, only: %i[new create]
   def new
     @post = Post.new
+    @categories = Category.all.map{ |cat| [cat.name, cat.id]}
   end
 
   def show
     @post = Post.find(params[:id])
     @author = Author.find(@post.author_id)
```

Comment:
Tutaj należałoby użyć helperów ActiveRecord.

skoro w modelu jest:
```ruby
class Post < ApplicationRecord
  belongs_to :author
```

to możemy używać `post.author`

```ruby
@author = @post.author
```

Reply:
Jednak skoro definiujemy 
```ruby
@post = Post.find(params[:id])
```

to do autora możemy dojść korzystając z `@post` i nie warto definiować `@autor` w kontrolerze

w posts/show.html.slim zamiast

```
  div.row
  - if author_signed_in?
    -if current_author.id == @author.id
```

lepiej

```
  div.row
  - if author_signed_in?
    -if current_author.id == @post.author.id
```